deployment.rst

```diff
@@ -1,8 +1,215 @@
+
+* Tagging a particular version of your code as a release in your source control
+  repository;
+* Creating a temporary staging area to build your updated setup "offline";
+* Running any tests available to ensure code and/or server stability;
+* Removal of any unnecessary files from the ``web/`` directory to keep your
+  production environment clean;
+* Clearing of external cache systems (like `Memcached`_ or `Redis`_).
+
+How to Deploy a Symfony Application
+-----------------------------------
+
+There are several ways you can deploy a Symfony application. Start with a few
+basic deployment strategies and build up from there.
+
+Basic File Transfer
+~~~~~~~~~~~~~~~~~~~
+
+The most basic way of deploying an application is copying the files manually
+via FTP/SCP (or similar method). This has its disadvantages as you lack control
+over the system as the upgrade progresses. This method also requires you
+to take some manual steps after transferring the files (see `Common Post-Deployment Tasks`_)
+
+Using Source Control
+~~~~~~~~~~~~~~~~~~~~
+
+If you're using source control (e.g. Git or SVN), you can simplify by having
+your live installation also be a copy of your repository. When you're ready
+to upgrade it is as simple as fetching the latest updates from your source
+control system.
+
+This makes updating your files *easier*, but you still need to worry about
+manually taking other steps (see `Common Post-Deployment Tasks`_).
+
+Using Build Scripts and other Tools
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+There are also tools to help ease the pain of deployment. Some of them have been
+specifically tailored to the requirements of Symfony.
+
+`Capistrano`_ with `Symfony plugin`_
+    `Capistrano`_ is a remote server automation and deployment tool written in Ruby.
+    `Symfony plugin`_ is a plugin to ease Symfony related tasks, inspired by `Capifony`_
+    (which works only with Capistrano 2 )
+
+`sf2debpkg`_
+    Helps you build a native Debian package for your Symfony project.
+
+`Magallanes`_
+    This Capistrano-like deployment tool is built in PHP, and may be easier
+    for PHP developers to extend for their needs.
+
+`Fabric`_
+    This Python-based library provides a basic suite of operations for executing
+    local or remote shell commands and uploading/downloading files.
+
+`Deployer`_
+    This is another native PHP rewrite of Capistrano, with some ready recipes for
+    Symfony.
+
+Bundles
+    There are some `bundles that add deployment features`_ directly into your
+    Symfony console.
+
+Basic scripting
+    You can of course use shell, `Ant`_ or any other build tool to script
+    the deploying of your project.
+
+Platform as a Service Providers
+    This guide includes detailed articles for some of the most well-known
+    Platform as a Service (PaaS) providers:
+
+    * :doc:`Microsoft Azure </deployment/azure-website>`
+    * :doc:`Heroku </deployment/heroku>`
+    * :doc:`Platform.sh </deployment/platformsh>`
```

Comment:
I see we have these down here - but I think they should have their own section above.